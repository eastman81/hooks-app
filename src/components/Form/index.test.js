import React from "react";
import Enzyme, { mount, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import ItemForm from ".";

Enzyme.configure({ adapter: new Adapter() });

describe("Form Component", () => {
  let wrapper = {};

  beforeEach(() => {
    wrapper = shallow(<ItemForm />);
  });

  it("renders without crashing", () => {
    expect(wrapper.exists()).toEqual(true);
  });

  it("should enter info with ENTER key", () => {
    const component = mount(<ItemForm />);
    
    component.find("#form").at(1).simulate("keydown", { keyCode: 13 });
    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it("updates on change", () => {
    const wrapper = mount(<ItemForm />);

    wrapper.find("select").simulate("change", { target: { value: "2" } });
    wrapper.find("input").simulate("change", { target: { value: "oranges" } });
    expect(wrapper.find("select").instance().value).toEqual("2");
    expect(wrapper.find("input").instance().value).toEqual("oranges");
  });
});
