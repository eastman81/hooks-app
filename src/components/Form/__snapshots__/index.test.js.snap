// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component should enter info with ENTER key 1`] = `
<ItemForm>
  <Form
    id="form"
    inline={false}
    onSubmit={[Function]}
  >
    <form
      className=""
      id="form"
      onSubmit={[Function]}
    >
      <FormRow>
        <div
          className="form-row"
        >
          <Col>
            <div
              className="col"
            >
              <FormControl
                as="select"
                id="formSelect"
                onChange={[Function]}
                value={1}
              >
                <select
                  className="form-control"
                  id="formSelect"
                  onChange={[Function]}
                  value={1}
                >
                  <option>
                    1
                  </option>
                  <option>
                    2
                  </option>
                  <option>
                    3
                  </option>
                  <option>
                    4
                  </option>
                  <option>
                    5
                  </option>
                  <option>
                    6
                  </option>
                  <option>
                    7
                  </option>
                  <option>
                    8
                  </option>
                  <option>
                    9
                  </option>
                  <option>
                    10
                  </option>
                  <option>
                    11
                  </option>
                  <option>
                    12
                  </option>
                </select>
              </FormControl>
            </div>
          </Col>
          <Col>
            <div
              className="col"
            >
              <FormControl
                id="formEnter"
                onChange={[Function]}
                placeholder="Item name"
                type="text"
                value=""
              >
                <input
                  className="form-control"
                  id="formEnter"
                  onChange={[Function]}
                  placeholder="Item name"
                  type="text"
                  value=""
                />
              </FormControl>
            </div>
          </Col>
        </div>
      </FormRow>
    </form>
  </Form>
</ItemForm>
`;
